webui/react/src/pages/TrialDetails/Profiles/utils.ts

```diff
@@ -0,0 +1,182 @@
+import { useEffect, useState } from 'react';
```

Comment:
**question:** we can cross that bridge if it does happen, but would any of this code be used anywhere else besides trial profiling?

Reply:
I'm not sure, in case we can just move on a lower level? But seems premature because this code is very specific now and if we're gonna consider chunking the app we'll have to move back.